### PR TITLE
Add additional rules to ruff and reenable black

### PR DESCRIPTION
**Pull request by MindTooth**
Could not understand why imports in #588 was not sorted. 😝 

Note: `config` is not known as an internal package.  Not sure if this is caused by `__init__.py`.

https://beta.ruff.rs/docs/rules/#isort-i

This part is just for testing:
```toml
[tool.ruff.isort]
known-first-party = ["src/config"]
```

### CODE REVIEW

**Comment by MindTooth** [comment]
However, setting it this way works:
```toml
[tool.ruff.isort]
known-first-party = ["config"]
```

But, I'm not sure why `ruff` would not detect this as first plass. 🤔

**Comment by MindTooth** [comment]
This escalated a bit.  Noticed that also Ruff don't replace black.

```
❯ ruff . --fix
scripts/download_weights.py:1:1: D100 Missing docstring in public module
scripts/download_weights.py:6:7: D101 Missing docstring in public class
scripts/download_weights.py:23:5: D103 Missing docstring in public function
src/core/api/schema.py:48:5: A003 Class attribute `id` is shadowing a Python builtin
src/core/api/schema.py:82:5: A003 Class attribute `id` is shadowing a Python builtin
src/core/api/schema.py:120:5: A003 Class attribute `id` is shadowing a Python builtin
src/core/api/schema.py:136:5: A003 Class attribute `id` is shadowing a Python builtin
src/core/api/schema.py:173:5: A003 Class attribute `id` is shadowing a Python builtin
src/core/api/schema.py:180:5: A003 Class attribute `id` is shadowing a Python builtin
src/core/repository/object.py:19:9: A003 Class attribute `list` is shadowing a Python builtin
src/core/repository/object.py:84:9: A003 Class attribute `list` is shadowing a Python builtin
src/core/repository/project.py:30:9: A003 Class attribute `list` is shadowing a Python builtin
src/core/repository/project.py:111:9: A003 Class attribute `list` is shadowing a Python builtin
src/core/repository/video.py:19:9: A003 Class attribute `list` is shadowing a Python builtin
src/core/repository/video.py:72:9: A003 Class attribute `list` is shadowing a Python builtin
src/tracing/tracker.py:31:24: A002 Argument `list` is shadowing a Python builtin
src/ui/projects/model.py:13:5: A003 Class attribute `id` is shadowing a Python builtin
src/ui/projects/model.py:51:5: A003 Class attribute `id` is shadowing a Python builtin
src/ui/projects/model.py:84:5: A003 Class attribute `id` is shadowing a Python builtin
src/ui/projects/model.py:144:5: A003 Class attribute `id` is shadowing a Python builtin
src/ui/projects/model.py:201:5: A003 Class attribute `id` is shadowing a Python builtin
src/ui/projects/model.py:228:5: A003 Class attribute `id` is shadowing a Python builtin
src/ui/projects/model.py:239:5: A003 Class attribute `id` is shadowing a Python builtin
tests/unit/detection/test_api.py:14:5: D103 Missing docstring in public function
Found 24 errors.
```

**Comment by tomrtk** [comment]
> This escalated a bit. Noticed that also Ruff don't replace black.
> 
> ```
> ❯ ruff . --fix
> scripts/download_weights.py:1:1: D100 Missing docstring in public module
> scripts/download_weights.py:6:7: D101 Missing docstring in public class
> scripts/download_weights.py:23:5: D103 Missing docstring in public function
> src/core/api/schema.py:48:5: A003 Class attribute `id` is shadowing a Python builtin
> src/core/api/schema.py:82:5: A003 Class attribute `id` is shadowing a Python builtin
> src/core/api/schema.py:120:5: A003 Class attribute `id` is shadowing a Python builtin
> src/core/api/schema.py:136:5: A003 Class attribute `id` is shadowing a Python builtin
> src/core/api/schema.py:173:5: A003 Class attribute `id` is shadowing a Python builtin
> src/core/api/schema.py:180:5: A003 Class attribute `id` is shadowing a Python builtin
> src/core/repository/object.py:19:9: A003 Class attribute `list` is shadowing a Python builtin
> src/core/repository/object.py:84:9: A003 Class attribute `list` is shadowing a Python builtin
> src/core/repository/project.py:30:9: A003 Class attribute `list` is shadowing a Python builtin
> src/core/repository/project.py:111:9: A003 Class attribute `list` is shadowing a Python builtin
> src/core/repository/video.py:19:9: A003 Class attribute `list` is shadowing a Python builtin
> src/core/repository/video.py:72:9: A003 Class attribute `list` is shadowing a Python builtin
> src/tracing/tracker.py:31:24: A002 Argument `list` is shadowing a Python builtin
> src/ui/projects/model.py:13:5: A003 Class attribute `id` is shadowing a Python builtin
> src/ui/projects/model.py:51:5: A003 Class attribute `id` is shadowing a Python builtin
> src/ui/projects/model.py:84:5: A003 Class attribute `id` is shadowing a Python builtin
> src/ui/projects/model.py:144:5: A003 Class attribute `id` is shadowing a Python builtin
> src/ui/projects/model.py:201:5: A003 Class attribute `id` is shadowing a Python builtin
> src/ui/projects/model.py:228:5: A003 Class attribute `id` is shadowing a Python builtin
> src/ui/projects/model.py:239:5: A003 Class attribute `id` is shadowing a Python builtin
> tests/unit/detection/test_api.py:14:5: D103 Missing docstring in public function
> Found 24 errors.
> ```

I would exclude the scripts folder. 

Also A003 errors could be fixed, but since it is class attributes I am not convinced it will be a issue, as it is accessed as eg. `job.id`. I would disable a003 for now.

**Review by tomrtk — APPROVED** [review]

**Comment by MindTooth** [comment]
Only one left:
```
src/tracing/tracker.py:31:24: A002 Argument `list` is shadowing a Python builtin
```

**Comment by tomrtk** [comment]
> Only one left:
> 
> ```
> src/tracing/tracker.py:31:24: A002 Argument `list` is shadowing a Python builtin
> ```

Rename it to something like `coordinate_list` would work

File: pyproject.toml [hunk 1]
```diff
@@ -93,28 +89,73 @@ write_to = "src/nina/version.py"
 # Tools Configuration #
 #######################
 
+[tool.black]
+line-length = 80
+target-version = ['py39']
+include = '\.pyi?$'
+exclude = '''
```
Comment (MindTooth) [inline pyproject.toml]: Maybe this could be rewritten to include only what we want to format, instead of exclude all cruft. 🤔

Comment (tomrtk) [reply pyproject.toml]: Maybe https://black.readthedocs.io/en/stable/usage_and_configuration/file_collection_and_discovery.html#gitignore

Comment (MindTooth) [reply pyproject.toml]: Solved it with:
```toml
[tool.black]
line-length = 80
target-version = ['py39']

include = '(src|tests)/.*\.pyi?$'
extend-exclude = 'src/vendor'
```

**Review by tomrtk — APPROVED** [review]

**Pull request merged**